app/controllers/api/storage.php

```diff
@@ -297,7 +299,7 @@
         $compressor = new GZIP();
         $device = Storage::getDevice('files');
 
-        if (!\file_exists($path)) {
+        if (!$device->exists($path)) {
```

Comment:
Is this method making an HTTP call on S3? can you tell how many call we make to S3 on different API endpoint to fulfill the user request?

Reply:
exists call will make 1 request and read will make 1 request, so this Preview will make two requests. Any suggestions on improvement?